src/test/java/bio/terra/workspace/service/resource/controlled/ControlledResourceServiceTest.java

```diff
@@ -278,6 +278,73 @@ public void createAiNotebookInstanceUndo() throws Exception {
                 userAccessUtils.defaultUserAuthRequest()));
   }
 
+  @Test
+  @DisabledIfEnvironmentVariable(named = "TEST_ENV", matches = bufferServiceDisabledEnvsRegEx)
+  public void createGetUpdateDeleteBqDataset() throws Exception {
+    Workspace workspace = reusableWorkspace();
+
+    String datasetId = "my_test_dataset";
+    String location = "us-central1";
+
+    ApiGcpBigQueryDatasetCreationParameters creationParameters =
+        new ApiGcpBigQueryDatasetCreationParameters().datasetId(datasetId).location(location);
+    ControlledBigQueryDatasetResource resource =
+        ControlledResourceFixtures.makeDefaultControlledBigQueryDatasetResource()
+            .workspaceId(workspace.getWorkspaceId())
+            .accessScope(AccessScopeType.ACCESS_SCOPE_SHARED)
+            .managedBy(ManagedByType.MANAGED_BY_USER)
+            .datasetName(datasetId)
+            .build();
+    ControlledBigQueryDatasetResource createdDataset =
+        controlledResourceService.createBqDataset(
+            resource,
+            creationParameters,
+            Collections.emptyList(),
+            userAccessUtils.defaultUserAuthRequest());
+    assertEquals(resource, createdDataset);
+
+    ControlledBigQueryDatasetResource getDataset =
```

Comment:
this name confuses me because it looks like a function name.

Reply:
good point, renamed for clarity